src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFChart.java

```diff
@@ -729,9 +770,11 @@ public CellReference setSheetTitle(String title, int column) {
     private void updateSheetTable(CTTable ctTable, String title, int index) {
         CTTableColumns tableColumnList = ctTable.getTableColumns();
         CTTableColumn column = null;
-        for( int i = 0; tableColumnList.getCount() < index; i++) {
+        int columnCount  = tableColumnList.getTableColumnList().size()-1;
```

Comment:
instead of starting loop  form 0 each time start it with table column size. previously each time we are adding new column from 0 inseatd from required index.